### PR TITLE
chore: move to swc/jest

### PR DESCRIPTION
**Pull request by riqwan**
what:

- moves most suites to use swc/jest
- fixes TS strictness that comes with using swc
- fixes unit test script to scope the shards correctly

### CODE REVIEW

File: packages/core/modules-sdk/src/medusa-module.ts [hunk 1]
```diff
@@ -576,4 +576,6 @@ export class MedusaModule {
 }
 
 global.MedusaModule ??= MedusaModule
-exports.MedusaModule = global.MedusaModule
+const GlobalMedusaModule = global.MedusaModule as typeof MedusaModule
+
+export { GlobalMedusaModule as MedusaModule }
```
Comment (riqwan) [inline packages/core/modules-sdk/src/medusa-module.ts:594]: @carlos-r-l-rodrigues can you check if this is correct.

File: scripts/run-workspace-unit-tests-in-chunks.sh [hunk 2]
```diff
@@ -21,5 +21,7 @@ for workspace in $(echo "$workspaces" | jq -r '.[]'); do
   filters+=" --filter=${workspace}"
 done
 
+command="yarn run test $filters"
+echo "$command"
```
Comment (carlos-r-l-rodrigues) [inline scripts/run-workspace-unit-tests-in-chunks.sh]: rm

**Review by carlos-r-l-rodrigues — COMMENTED** [review]
This looks great man.

File: scripts/run-workspace-unit-tests-in-chunks.sh (hunk 2 above)
Comment (riqwan) [inline scripts/run-workspace-unit-tests-in-chunks.sh]:
```suggestion
```

**Review by sradevski — APPROVED** [review]
Nice, did you notice any speed improvements in the pipeline?

**Comment by riqwan** [comment]
@sradevski I didn't think there would be since the tests affected were fairly small, but I remember being a decent improvement. Lets find out with the new run 😄

**Comment by riqwan** [comment]
@sradevski looks like there a decent improvement

**Comment by thetutlage** [comment]
Breath of fresh air 💨

File: package.json [hunk 3]
```diff
@@ -58,7 +58,7 @@
     "lint:path": "eslint --ignore-path .eslintignore --ext .js,.ts,.tsx",
     "prettier": "prettier",
     "jest": "jest",
-    "test": "turbo run test --concurrency=50% --no-daemon --no-cache --force --filter='./packages/*' --filter='./packages/core/*' --filter='./packages/cli/*' --filter='./packages/modules/*' --filter='./packages/modules/providers/*'",
+    "test": "turbo run test --concurrency=50% --no-daemon --no-cache --force",
```
Comment (adrien2p) [inline package.json:61]: Q: I remember we added those filters to be sure it only runs the unit tests from packages but that the filter did not allow for deep regexp, are we sure only the unit tests are ran from the packages directory here?

File: packages/core/modules-sdk/src/medusa-module.ts (hunk 1 above)
Comment (adrien2p) [reply packages/core/modules-sdk/src/medusa-module.ts:594]: should work the same 👍

**Review by adrien2p — APPROVED** [review]
Looks good to me 💪

File: package.json (hunk 3 above)
Comment (riqwan) [reply package.json:61]: the .sh script will filter to run all packages in the workspace. Which ones need to be filtered out?

Comment (adrien2p) [reply package.json:61]: oh ok perfect

**Pull request merged**